ksql-engine/src/main/java/io/confluent/ksql/physical/PhysicalPlanBuilder.java

```diff
@@ -278,8 +278,10 @@ private QueryMetadata buildPlanForStructuredOutputNode(
 
   private void sinkSetUp(final KsqlStructuredDataOutputNode outputNode,
                          final StructuredDataSource sinkDataSource) {
-    if (updateMetastore && outputNode.isDoCreateInto()) {
-      metaStore.putSource(sinkDataSource.cloneWithTimeKeyColumns());
+    if (outputNode.isDoCreateInto()) {
```

Comment:
The engine's `tryExecute` now follows the same code paths as `execute`, which means it comes into this method with a proper output node. (The old code followed very different code paths in here between validation and execution).  

Without this change the validation below was incorrectly being invoked where `isDoCreateInto` was true, but `updateMetastore` was false.  With the new logic it does the right thing.